### DISCUSSION TITLE
How to set a default value for timestamp column?

### DISCUSSION THREAD

**Discussion by frederikhors** [SeaQL/sea-orm#1035] [Q&A] [answered]
Can you help me understand how to set a default value for `timestamp` column?

```rust
ColumnDef::new(Player::CreatedAt)
    .timestamp()
    .not_null()
    .default(/*WHAT?*/)
```

**Comment by frederikhors** [comment, marked as answer]
I found out how (for now): https://github.com/SeaQL/sea-query/discussions/346.

**Reply by ikrivosheev** [reply]
@frederikhors hello! We are working on this)

**Reply by billy1624** [reply]
Tracking PR and issue:
- https://github.com/SeaQL/sea-query/pull/428
- https://github.com/SeaQL/sea-query/issues/347

**Reply by christopherc1331** [reply]
Thanks for the link to the pull request! (I see that it was merged :D) 

@frederikhors  Now we can solve this with something like this: 

`.col(ColumnDef::new(Users::UpdatedTS).timestamp().default(SimpleExpr::Keyword(Keyword::CurrentTimestamp)).not_null())`

**Comment by barsdeveloper** [comment]
I am creating my tables from the Model using schema.create_table_from_entity
Is there a way to set the current timestamp as a default value in the model?
```Rust
#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "configuration")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    #[sea_orm(default_value = CURRENT_TIMESTAMP)]
    pub some_timestamp: String,
}
```

This does not work because it expects some literal value.

**Reply by YCCDSZXH** [reply]
I found this code works..
```rust
#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "configuration")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    #[sea_orm(default_expr = "Expr::current_timestamp()")]
    pub some_timestamp: String,
}
```
This code also works, and it generate sql with expected type
```rust
chrono::DateTime<Utc>
#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "configuration")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    #[sea_orm(default_expr = "Expr::current_timestamp()")]
    pub some_timestamp: chrono::DateTime<Utc>,
}
```
```sql
`created_at` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
